RecoEgamma/EgammaHLTProducers/plugins/EgammaHLTPixelMatchVarProducer.cc

```diff
@@ -41,7 +41,24 @@ namespace {
       int layerOffset = 3;
       if (seed.subDet(hitNr) == PixelSubdetector::PixelEndcap)
         layerOffset += 4;
-      int layerBit = 0x1 << layerOffset << seed.layerOrDiskNr(hitNr);
+
+      int layerOrDiskNr = seed.layerOrDiskNr(hitNr);
+      if (layerOrDiskNr == 0 || layerOrDiskNr > 4) {
```

Comment:
for my education what happens in the phase-2 case? You don't foresee to make use of the full FPix ?

Reply:
ah thanks, I had understood from our brief conversation that we were still 4 layers in phase-II and of course we dont, sorry my bad. 

Right now we dont take advantage of the full fpx only 1-4 and we should review this as it'll be important at high eta. But this requires a reworking of this code and more reason to have the warning in there to cause problems. 

Thanks for pointing it out

Reply:
> ah thanks, I had understood from our brief conversation that we were still 4 layers in phase-II and of course we dont, sorry my bad.

In phase-2 we'll still have 4 layers in BPix, but 12 disks in the endcaps:

![image](https://github.com/user-attachments/assets/3f6d5a0a-0935-4d5b-9c54-43289962c1cf)

sorry it was probably me not being clear during our conversation.
So, IIUC, you don't think this 4 should be a 12 for the time being until there's more code rework, right?